### ISSUE TITLE
None of the keys work.

### ISSUE THREAD

**Issue by Federino** [tompi/cheapino#36]
Hi!

I am having trouble with the cheapini v2 build. I soldered all diodes with the MCU and the ethernet port. After that i decided to solder one switch and test if everything I did works before soldering everything else, but I don't get any output after pressing the switch.

I tryed the default software and the miryoku.uf2 as well, but none of them work.

If i go to vial.rock website it detects the pcb as Cheapino, but when I try the matrix test the button doesn't work.

Do I need to solder all the components before trying?
I am sending pictures as well so you can see the PCB.
![IMG_20240514_184231](https://github.com/tompi/cheapino/assets/160331519/6ddba8cd-5381-4677-bac9-2f08e5f695d0)
![IMG_20240514_184241](https://github.com/tompi/cheapino/assets/160331519/3dedece6-1b5a-4bc9-b6e0-8c616b4dfb72)
![IMG_20240514_184304](https://github.com/tompi/cheapino/assets/160331519/ee2cb2d2-773d-4d34-9fb9-8faee7098e8a)
![IMG_20240514_184341](https://github.com/tompi/cheapino/assets/160331519/9adca515-cfd4-43d9-bd88-5713e8eeb471)

I tryed the troubleshooting points, but I can't seem to make it work.

I also used a tester to check the diodes and they seem fine. I also soldered two switches in order to check if one of them is faulty.

If you need more information please let me know. Thank you!

**Comment by Federino** [comment]
Update: It seems like 4 of the keys work. Maybe I have the other diodes not polarised correctly, I will try and resolder everything this afternoon.

![IMG_20240514_194053](https://github.com/tompi/cheapino/assets/160331519/847b9c0d-a284-4bc4-8ee2-ff39a59200e0)

**Comment by tompi** [comment]
Hi. You definitely have some bad(a few cold) solder joints which could cause problems.

Try removing excess solder and reflow. When you got it looking nice, let me know if there is still problems and we can debug a specific trace with a multimeter(but I think its solder joints now...)

![image](https://github.com/tompi/cheapino/assets/231717/e0587102-2697-419f-886a-58c33c2e7952)

**Comment by Federino** [comment]
Hi!

I'm really sorry for the late reply, but I didn't have time to fix it until today, and I MADE IT WORK!

It turns out that I had a few things soldered incorrectly:

1) I tryed to use the female conector for the MCU and I did a bad insert of the male "sticks" and they ended up not making contact correctly, which is why none of the keys worked. To solve this, I resoldered the entire chip and It worked perfectly.

2) After I managed to fix that I found out that the top row of the left part was not working either. After some debug of the PCB I found that I did not bridge correctly the right side of the PCB, so I redid that part as well.

After that I managed to make everything work.

I thank you so much for the support and this amazing product!
